mqtt/mqtt-broker/src/auth/mod.rs

```diff
@@ -84,3 +84,9 @@ impl PartialEq<ClientId> for Identity {
         self.as_str() == other.as_str()
     }
 }
+
+impl PartialEq<String> for &Identity {
```

Comment:
```suggestion
impl PartialEq<T: AsRef<str>> for &Identity {
```
it will make ergonomics better

Reply:
impl PartialEq<T> for &Identity will be derived automatically

Reply:
Do we actually need this implementation? As far as I can see, you use it in one place. Should we create a ClientId instead of String?
```rust
let actor_id = ClientId::from(format!("{}/{}", self.iothub_id, activity.client_id()));
if identity == actor_id {
...
}